portal/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -1,15 +1,55 @@
 package cromwell.pipeline.service
 
 import cromwell.pipeline.datastorage.dao.repository.UserRepository
+import cromwell.pipeline.datastorage.dto.user.{ PasswordUpdateRequest, UserUpdateRequest }
 import cromwell.pipeline.datastorage.dto.{ UserId, UserNoCredentials }
+import cromwell.pipeline.utils.StringUtils
 
 import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.{ Random }
 
 class UserService(userRepository: UserRepository)(implicit executionContext: ExecutionContext) {
+  import StringUtils._
 
   def deactivateUserById(userId: UserId): Future[Option[UserNoCredentials]] =
     for {
       _ <- userRepository.deactivateUserById(userId)
       user <- userRepository.getUserById(userId)
     } yield user.map(UserNoCredentials.fromUser)
+
+  def updateUser(userId: String, request: UserUpdateRequest): Future[Int] =
+    userRepository
+      .getUserById(UserId(userId))
+      .flatMap(
+        user =>
+          user.map(
+            user =>
```

Comment:
Giving the same name to different variables a bit confusing.

Reply:
Done